### ISSUE TITLE
`strutils.plural`  does not handle acronyms correctly

### ISSUE THREAD

**Issue by mhucka** [mahmoud/boltons#291]
If the word to be pluralized is an acronym like "DOI", then `plural` will match case and produce "DOIS", which is not the desired result in many (most?) cases. A better result would be "DOIs", to match its use in sentences such as "There are a total of 53 DOIs associated with this journal".

I realize acronyms are especially challenging, because it's basically impossible for `plural` to recognize every conceivable acronym. Perhaps a better solution would be to have some kind of option or override on the case-matching part of the code, such that the user could say "don't match case".

**Comment by mahmoud** [comment]
Good point, I'd take a PR for that if you want to take a swing at it. :)
